prefork/prefork.go

```diff
@@ -0,0 +1,182 @@
+package prefork
+
+import (
+	"flag"
+	"net"
+	"os"
+	"os/exec"
+	"runtime"
+
+	"github.com/valyala/fasthttp"
+	"github.com/valyala/fasthttp/reuseport"
+)
+
+const preforkChildFlag = "-prefork-child"
+const defaultNetwork = "tcp4"
+
+// Prefork implements fasthttp server prefork
+//
+// Preforks master process (with all cores) between several child processes
+// increases performance significantly, because Go doesn't have to share
+// and manage memory between cores
+//
+// WARNING: using prefork prevents the use of any global state!
+// Things like in-memory caches won't work.
+type Prefork struct {
+	// The network must be "tcp", "tcp4" or "tcp6".
+	//
+	// By default is "tcp4"
+	Network string
+
+	// Flag to use a listener with reuseport, if not a File Listener will be used
+	//
+	// It's disabled by default
+	Reuseport bool
+
+	ServeFunc         func(ln net.Listener) error
+	ServeTLSFunc      func(ln net.Listener, certFile, keyFile string) error
+	ServeTLSEmbedFunc func(ln net.Listener, certData, keyData []byte) error
+
+	ln    net.Listener
+	files []*os.File
+}
+
+func init() { //nolint:gochecknoinits
+	flag.Bool(preforkChildFlag[1:], false, "is a child process")
```

Comment:
Why is this still here? It doesn't do anything?

Reply:
Example:

```go
package main

import (
	"flag"
	"log"
	"os"

	"github.com/valyala/fasthttp"
	"github.com/valyala/fasthttp/prefork"
)

var (
	addr = flag.String("addr", ":8080", "TCP address to listen to")
)

func requestHandler(ctx *fasthttp.RequestCtx) {
	ctx.WriteString("Hello World")
}

func main() {
	flag.Parse()

	s := &fasthttp.Server{Handler: requestHandler}
	ps := prefork.New(s)

	log.Printf("Running on %s (pid: %d)", *addr, os.Getpid())
	if err := ps.ListenAndServe(*addr); err != nil {
		log.Fatalf("Error in ListenAndServe: %s", err)
	}
}
```

If i not define the flag `prefork-child`, i will take the following error:
```bash
2020/02/12 12:35:21 Running on :8080 (pid: 12547)
flag provided but not defined: -prefork-child
Usage of /var/folders/kl/zdgwjx956cv0yw1pt89zzxt80000gp/T/go-build831309057/b001/exe/helloworldserver:
  -addr string
        TCP address to listen to (default ":8080")
```

Reply:
Only in cases that the user create their own flags

Reply:
Ok then it makes sense. But please add a comment explaining why cause right now someone reading the code will not understand why that is there.

Reply:
Done